diff --git a/py-compile b/py-compile
--- a/py-compile
+++ b/py-compile
@@ -119 +119 @@
-import sys, os, py_compile, imp
+import sys, os, py_compile, importlib.util
@@ -132,2 +132,2 @@
-    if hasattr(imp, 'get_tag'):
-        py_compile.compile(filepath, imp.cache_from_source(filepath), path)
+    if hasattr(importlib.util, 'cache_from_source'):
+        py_compile.compile(filepath, importlib.util.cache_from_source(filepath), path)
@@ -140 +140 @@
-import sys, os, py_compile, imp
+import sys, os, py_compile, importlib.util
@@ -156,2 +156,2 @@
-    if hasattr(imp, 'get_tag'):
-        py_compile.compile(filepath, imp.cache_from_source(filepath, False), path)
+    if hasattr(importlib.util, 'cache_from_source'):
+        py_compile.compile(filepath, importlib.util.cache_from_source(filepath, False), path)
