--- a/gio/gosxcontenttype.c
+++ b/gio/gosxcontenttype.c
@@ -58,11 +58,16 @@ create_cstr_from_cfstring (CFStringRef str)
 {
+  CFIndex length;
+  CFIndex maxlen;
+  gchar *buffer;
+  Boolean success;
+
   g_return_val_if_fail (str != NULL, NULL);
 
-  CFIndex length = CFStringGetLength (str);
-  CFIndex maxlen = CFStringGetMaximumSizeForEncoding (length, kCFStringEncodingUTF8);
-  gchar *buffer = g_malloc (maxlen + 1);
-  Boolean success = CFStringGetCString (str, (char *) buffer, maxlen,
-                                        kCFStringEncodingUTF8);
+  length = CFStringGetLength (str);
+  maxlen = CFStringGetMaximumSizeForEncoding (length, kCFStringEncodingUTF8);
+  buffer = g_malloc (maxlen + 1);
+  success = CFStringGetCString (str, (char *) buffer, maxlen,
+                                kCFStringEncodingUTF8);
   CFRelease (str);
   if (success)
