--- a/configure.ac
+++ b/configure.ac
@@ -2051,6 +2051,9 @@
     	i386) 
     		MACOSX_DEFAULT_ARCH="i386" 
     		;;
+    	arm64)
+    		MACOSX_DEFAULT_ARCH="arm64"
+    		;;
     	ppc) 
     		MACOSX_DEFAULT_ARCH="ppc" 
     		;;
@@ -2063,6 +2066,9 @@
     	i386) 
     		MACOSX_DEFAULT_ARCH="x86_64" 
     		;;
+    	arm64)
+    		MACOSX_DEFAULT_ARCH="arm64"
+    		;;
     	ppc) 
     		MACOSX_DEFAULT_ARCH="ppc64" 
     		;;
--- a/configure
+++ b/configure
@@ -8459,6 +8459,9 @@
     	i386)
     		MACOSX_DEFAULT_ARCH="i386"
     		;;
+    	arm64)
+    		MACOSX_DEFAULT_ARCH="arm64"
+    		;;
     	ppc)
     		MACOSX_DEFAULT_ARCH="ppc"
     		;;
@@ -8471,6 +8474,9 @@
     	i386)
     		MACOSX_DEFAULT_ARCH="x86_64"
     		;;
+    	arm64)
+    		MACOSX_DEFAULT_ARCH="arm64"
+    		;;
     	ppc)
     		MACOSX_DEFAULT_ARCH="ppc64"
     		;;
